### ISSUE TITLE
Modified Gravity option in Cosmology Calculator 

### ISSUE THREAD

**Issue by agnesemenaite** [LSSTDESC/CCL#976]
Currently, pyccl CosmologyCalculator function does not allow specification of modified gravity parameters, which can lead to an error in Cls, even if nonlinear power spectrum is calculated accurately. 

Below is an illustration of the issue for mu-sigma model with mu0 and sigma0 both set to 0.2 in 'Standard Cosmology' case where the Cls (for shear) are constructed with the standard pyccl Cosmology object. The calculator uses the resulting nonlinear power spectra and matching cosmology, but with no modified gravity parameter specification:

[cl_mg_calculator.pdf](https://github.com/LSSTDESC/CCL/files/9271088/cl_mg_calculator.pdf)
[ps_mg_calculator.pdf](https://github.com/LSSTDESC/CCL/files/9271098/ps_mg_calculator.pdf)

**Comment by nikfilippas** [comment]
I managed to reproduce that using the following code:
``` python
import pyccl as ccl
import numpy as np
import matplotlib.pyplot as plt


# set up a MG cosmology
cosmo_MG = ccl.Cosmology(Omega_c=0.25, Omega_b=0.05, h=0.67, n_s=0.96,
                         A_s=2.34e-9, mu_0=0.5, sigma_0=0.5,
                         transfer_function="boltzmann_isitgr")
cosmo_MG.compute_nonlin_power()
pk2d = cosmo_MG.get_nonlin_power()


# copy it into a calculator
a, lk, pk = pk2d.get_spline_arrays()
pk_nonlin = {"a": a, "k": np.exp(lk), "delta_matter:delta_matter": pk}
cosmo = ccl.CosmologyCalculator(Omega_c=0.25, Omega_b=0.05, h=0.67, n_s=0.96,
                                A_s=2.34e-9, pk_nonlin=pk_nonlin)


# get the MG Cells
ell = np.geomspace(2, 2000, 128)
tr_MG = ccl.CMBLensingTracer(cosmo_MG, 1100.)
cl0 = ccl.angular_cl(cosmo_MG, tr_MG, tr_MG, ell)

# get the calculator Cells
cosmo.compute_growth()
tr = ccl.CMBLensingTracer(cosmo, 1100.)
cl1 = ccl.angular_cl(cosmo, tr, tr, ell)

plt.loglog(ell, cl0, label="cosmo MG")
plt.loglog(ell, cl1, label="cosmo calc")


# simple cosmology without MG
cosmo_test = ccl.Cosmology(Omega_c=0.25, Omega_b=0.05, h=0.67, n_s=0.96, A_s=2.34e-9,
                           transfer_function="boltzmann_isitgr")
cosmo_test.compute_nonlin_power()
tr_test = ccl.CMBLensingTracer(cosmo_test, 1100.)
cl2 = ccl.angular_cl(cosmo_test, tr_test, tr_test, ell)
plt.loglog(ell, cl2, label="cosmo no MG")

plt.legend()
plt.xlabel("ell")
plt.ylabel("cell")
```
which gives this plot
![Screenshot from 2022-08-06 00-13-00](https://user-images.githubusercontent.com/22447385/183195694-13c315a7-f462-4170-aecd-bd7413a377b2.png)

**Comment by damonge** [comment]
I am happy to move ahead with the fix in #977, but with the caveat that in v3 this will probably be deprecated. The calculator mode was intended to allow for these kinds of things in a much more general way. Users will pass a set of auto and cross power spectra between the matter overdensity and the weyl potential, and the C_ells will be computed by integrating the appropriate Pk. This will allow full freedom to include models with arbitrary scale-dependent growth
